monitor/uidmonitor/processor.go

```diff
@@ -0,0 +1,398 @@
+			return err
+		}
+
+		defaultIP, _ := runtimeInfo.DefaultIPAddress()
+
+		zap.L().Error("Starting ", zap.String("contextID", contextID), zap.String("Publishing", publishedContextID))
+		if perr := s.puHandler.HandlePUEvent(publishedContextID, monitor.EventStart); perr != nil {
+			zap.L().Error("Failed to activate process", zap.Error(perr))
+			return perr
+		}
+
+		err = s.processLinuxServiceStart(eventInfo, runtimeInfo)
+
+		if err != nil {
+			zap.L().Error("ProcessLInuxServiceStart", zap.Error(err))
+			return err
+		}
+
+		s.collector.CollectContainerEvent(&collector.ContainerRecord{
+			ContextID: contextID,
+			IPAddress: defaultIP,
+			Tags:      runtimeInfo.Tags(),
+			Event:     collector.ContainerStart,
+		})
+		entry := &putoPidEntry{
+			Info:               runtimeInfo,
+			publishedContextID: publishedContextID,
+		}
+		entry.pidlist = make(map[string]bool, 20)
+		entry.pidlist[eventInfo.PID] = true
+		s.puToPidEntry.Add(contextID, entry)
+		s.pidToPU.Add(eventInfo.PID, contextID)
+		// Store the state in the context store for future access
+		zap.L().Error("ContextID", zap.String("eventInfo.PID", eventInfo.PID), zap.String("eventInfo.OUID", contextID))
+		return s.contextStore.StoreContext(contextID, &StoredContext{
+			EventInfo: eventInfo,
+			MarkVal:   runtimeInfo.Options().CgroupMark,
+		})
+
+	}
+	zap.L().Error("Adding to existing session", zap.String("contextID", contextID))
+	pids.(*putoPidEntry).pidlist[eventInfo.PID] = true
+	s.pidToPU.Add(eventInfo.PID, eventInfo.PUID)
+	err = s.processLinuxServiceStart(eventInfo, pids.(*putoPidEntry).Info)
+	return err
+
+}
+
+// Stop handles a stop event
+func (s *UIDProcessor) Stop(eventInfo *rpcmonitor.EventInfo) error {
+
+	contextID, err := s.generateContextID(eventInfo)
+	if err != nil {
+		return err
+	}
+	if contextID == "/trireme" {
+		return nil
+	}
+	strtokens := strings.Split(contextID, "/")
+	contextID = "/" + strtokens[len(strtokens)-1]
+	zap.L().Error("UID STOP EventInfo", zap.String("EventInfo.PUID", eventInfo.PUID),
+		zap.String("EventInfo.PID", eventInfo.PID),
+		zap.String("EventType", string(eventInfo.EventType)),
+		zap.Bool("HostService", eventInfo.HostService),
+		zap.String("CGROUP", eventInfo.Cgroup),
+		zap.String("contextID", contextID),
+	)
+	s.Lock()
+	defer s.Unlock()
+	stoppedpid := strtokens[len(strtokens)-1]
+	if puid, err := s.pidToPU.Get(stoppedpid); err == nil {
+		eventInfo.PUID = puid.(string)
+	}
+
+	var publishedContextID string
+	zap.L().Error("167", zap.String("contextID", contextID))
+	if pidlist, err := s.puToPidEntry.Get(contextID); err == nil {
+		publishedContextID = pidlist.(*putoPidEntry).publishedContextID
+		if len(pidlist.(*putoPidEntry).pidlist) > 1 {
+			zap.L().Error("Length of PIDLIST", zap.Int("Length", len(pidlist.(*putoPidEntry).pidlist)))
+			return nil
+		}
+	}
+	if len(strtokens) == 1 && contextID == "trireme" {
+		return nil
+	}
+
+	hperr := s.puHandler.HandlePUEvent(publishedContextID, monitor.EventStop)
```

Comment:
directly return your error, and if you want to keep your loh (that has an extra space), don’t allocate a new error